### PR TITLE
Added TECA Installation documentation

### PR DESCRIPTION
**Pull request by elbashandy**

### CODE REVIEW

**Comment by elbashandy** [comment]
The installation guide was tested on an Ubuntu docker container.

**Comment by taobrienlbl** [comment]
This is wonderful!  Do you think it makes sense to include here documentation of the setup.py install?  Also, is there a simple way for me to view the Sphinx output?

**Comment by burlen** [comment]
@taobrienlbl when this is all setup and working correctly updates will be automatic. we will use this branch for documentation changes and pushes should be visible at https://teca.readthedocs.io/en/sphinx_documentation/

@elbashandy  there seem to be some missing files, or our conf.py isn't quite right
see: https://readthedocs.org/projects/teca/builds/9249475/

**Comment by elbashandy** [comment]
@taobrienlbl The way I used to visualize it was to run `make html` in the `TECA/docs` folder to generate html files. You will find the entry html file is `TECA/docs/_build/index.html`. You are right, we should add setup.py as well.

@burlen amazing! I thought `docs` needs to be merged to master to be visualized in readthedocs. Thanks for letting me know. Will see why we're getting this error.

**Comment by burlen** [comment]
@elbashandy also please put this new sphinx stuff in a new folder named `rtd-docs`, to keep it separate from our older stuff in the `docs` folder. Thanks!

**Comment by elbashandy** [comment]
@burlen the old stuff are in the `doc` folder (not `docs`). I know having both is quite confusing and we should rename them. I will change `docs` to `rtd-docs`

**Comment by burlen** [comment]
@taobrienlbl @elbashandy I converted our user guide, and configured rtd such that this branch is automatically built and deployed to
https://teca.readthedocs.io/en/sphinx_documentation/

**Comment by elbashandy** [comment]
@burlen Amazing!! Thank you

File: rtd-docs/installation.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    $ export PYTHONPATH=${TECA_DIR}/build/lib:${PYTHONPATH}
+    $ export LD_LIBRARY_PATH=${TECA_DIR}/build/lib:${LD_LIBRARY_PATH}
+
+If everything went well we can confirm that it worked by trying::
+
+    $ python3 -c "from teca import *"
+
+Now we're ready to go!
+
+
```
Comment (taobrienlbl) [inline rtd-docs/installation.rst]: Adding a reminder here about `setup.py` installation instructions.

**Comment by taobrienlbl** [comment]
This is looking awesome!  I'm really glad to see these docs online like this.  Nice work.

**Comment by burlen** [comment]
Don't delete this branch, we will use it for future mods to the rtd site

**Pull request merged**